odh-model-controller/controllers/storageconfig_controller.go

```diff
@@ -0,0 +1,214 @@
+		}
+	}
+	desiredStorageSecret := newStorageSecret(dataConnectionSecretsList)
+	desiredStorageSecret.Name = storageSecretName
+	desiredStorageSecret.Namespace = secret.Namespace
+	desiredStorageSecret.Labels = map[string]string{}
+	desiredStorageSecret.Labels["opendatahub.io/managed"] = "true"
+
+	foundStorageSecret := &corev1.Secret{}
+	justCreated := false
+	err = r.Get(ctx, types.NamespacedName{
+		Name:      desiredStorageSecret.Name,
+		Namespace: secret.Namespace,
+	}, foundStorageSecret)
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			log.Info("Creating Storage Config Secret")
+
+			// Create the secret  in the Openshift cluster
+			err = r.Create(ctx, desiredStorageSecret)
+			if err != nil && !apierrs.IsAlreadyExists(err) {
+				log.Error(err, "Unable to create the Secret")
+				return err
+			}
+			justCreated = true
+		} else {
+			log.Error(err, "Unable to fetch the Storage Config Secret")
+			return err
+		}
+	}
+
+	// Reconcile the Storage Config Secret if it has been manually modified
+	if !justCreated && !CompareStorageSecrets(*desiredStorageSecret, *foundStorageSecret) {
+		log.Info("Reconciling Storage Config Secret")
+
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			// Get the last Storage Config revision
+			if err := r.Get(ctx, types.NamespacedName{
+				Name:      desiredStorageSecret.Name,
+				Namespace: secret.Namespace,
+			}, foundStorageSecret); err != nil {
+				return err
+			}
+			// Reconcile labels and spec field
+			foundStorageSecret.Data = desiredStorageSecret.Data
+			foundStorageSecret.ObjectMeta.Labels = desiredStorageSecret.ObjectMeta.Labels
+			return r.Update(ctx, foundStorageSecret)
+		})
+		if err != nil {
+			log.Error(err, "Unable to reconcile the Storage Config Secret")
+			return err
+		}
+	}
+
+	return nil
+}
+
+// ReconcileStorageSecret will manage the creation, update and deletion of the Storage Config Secret
+func (r *StorageSecretReconciler) ReconcileStorageSecret(
+	secret *corev1.Secret, ctx context.Context) error {
+	return r.reconcileSecret(secret, ctx, newStorageSecret)
+}
+
+// reconcileOpenDataHubSecrets will filter out all secrets that are not managed by the ODH
+func reconcileOpenDataHubSecrets() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			objectLabels := e.Object.GetLabels()
+			if objectLabels["opendatahub.io/managed"] == "true" {
+				return true
+			}
+			return false
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			objectLabels := e.Object.GetLabels()
+			if objectLabels["opendatahub.io/managed"] == "true" {
+				return true
+			}
+			return false
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			objectLabels := e.Object.GetLabels()
+			if objectLabels["opendatahub.io/managed"] == "true" {
+				return true
+			}
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			objectOldLabels := e.ObjectOld.GetLabels()
+			objectNewLabels := e.ObjectOld.GetLabels()
```

Comment:
Is this supposed to be ObjectOld as well?

Reply:
Good catch